File: 438_Find_All_Anagrams_in_a_String.js
var findAnagrams = function(s, p) {
    let n = p.length
    p = p.split("").sort().join()
    let ans = []
    for(let i = 0; i <= s.length - n; i++){
        let substr = s.substring(i, i + n)
        substr = substr.split("").sort().join()
        if(substr == p) ans.push(i)
    }
    return ans
};

File: 438_Find_All_Anagrams_in_a_String.cpp
class Solution {
public:
    vector<int> findAnagrams(string s, string p) {
        vector<int> ans;
        sort(p.begin(),p.end());
        int m = p.length();
        int n = s.length();
        int i = 0;
        while(i <= n - m){
            string temp = s.substr(i,m);
            sort(temp.begin(),temp.end());
            if(temp == p) ans.push_back(i);
            i++;
        }
        return ans;
    }
};
